base/abstractarraymath.jl

```diff
@@ -48,9 +48,11 @@ _sub(t::Tuple, s::Tuple) = _sub(tail(t), tail(s))
 """
     dropdims(A; dims)
 
-Remove the dimensions specified by `dims` from array `A`.
-Elements of `dims` must be unique and within the range `1:ndims(A)`.
-`size(A,i)` must equal 1 for all `i` in `dims`.
+Return an array with the same data as `A`, but with the dimensions specified by
+`dims` removed. The two arrays may share the same underlying data, such that the
+result is mutable if and only if `A` is mutable, and setting elements of one
+alters the values of the other. Elements of `dims` must be unique and within
+the range `1:ndims(A)`. `size(A,i)` must equal 1 for all `i` in `dims`.
```

Comment:
Can I suggest moving the story about sharing data a bit later? I think that the basic fact that it drops dimensions of size 1 gets a bit lost here. 
```suggestion
Return an array with the same data as `A`, but with the dimensions specified by
`dims` removed. `size(A,d)` must equal 1 for every `d` in `dims`,
and repeated dimensions or numbers outside `1:ndims(A)` are forbidden.

The result may share the same underlying data as `A`, such that the
result is mutable if and only if `A` is mutable, and setting elements of one
alters the values of the other.
```

Reply:
That's fine for me. The heavy emphasis on shared memory right in the beginning is taken from the `reshape` docstring.